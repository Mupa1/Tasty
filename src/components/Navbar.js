import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <header className="bg-primary">
    <nav className="text-center">
      <Link to="/">
        <h1 className="text-white">Tasty Recipes</h1>
      </Link>
    </nav>
  </header>
);

export default Navbar;
