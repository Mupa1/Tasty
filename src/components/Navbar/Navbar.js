import React from 'react';
import { Link, BrowserRouter } from 'react-router-dom';

import './Navbar.module.css';

const Navbar = () => (
  <header className="bg-success text-center">
    <nav data-testid="navbar">
      <BrowserRouter>
        <Link to="/">
          <h1 className="text-white">Tasty Recipes</h1>
        </Link>
      </BrowserRouter>
    </nav>
  </header>
);

export default Navbar;
