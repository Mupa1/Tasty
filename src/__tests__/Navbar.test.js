import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import Navbar from '../components/Navbar/Navbar';

it('Navbar component renders correctly', () => {
  const tree = renderer
    .create(<Navbar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Navbar', () => {
  test('contains Tasty Recipes logo', () => {
    render(<Navbar />);
    expect(screen.getByText('Tasty Recipes')).toBeInTheDocument();
  });
});
