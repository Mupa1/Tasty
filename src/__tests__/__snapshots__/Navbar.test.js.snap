// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders correctly 1`] = `
<header
  className="bg-success text-center"
>
  <nav>
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="text-white"
      >
        Tasty Recipes
      </h1>
    </a>
  </nav>
</header>
`;
